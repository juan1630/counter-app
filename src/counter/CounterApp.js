import React, { useState } from 'react';
import PropTypes  from 'prop-types'

// funcional component

const CounterApp = ({value  = 2} ) => {

  const [counter, setCounter] = useState( value );
// retorn un arreglo
  const handleadd = ()=>  {
    // console.log(event);

    // setCounter(  counter + 1);
    setCounter( (c) => c +1 );
    // estas son las 2 maneras de manejar el counter y su estado
    // cambiamos el valor del counter
    // solo las funciones que estan dedicadas a cambiar el state
  }

  const handleRest = () => {
    setCounter( c => c - 1 )
  }

  const handleReste = () => {
      setCounter( value );
  }


  return (
    <>
    <h1> Counter App </h1>
    <h2>{counter}</h2>

    <button  onClick={ handleadd } > +1 </button>
    <button  onClick={ handleRest } > -1 </button>
    <button  onClick={ handleReste } > RESET </button>

  </>
  )
}

// refactor de la funcion
// hace referencia  a la funcion que se declaro arriba


CounterApp.propTypes = {
  value:PropTypes.number
}
export default CounterApp;
