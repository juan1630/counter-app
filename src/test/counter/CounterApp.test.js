import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import CounterApp from '../../counter/CounterApp';

describe('Son las pruebas del componente <CounterApp/>', () => {

    test('Debe de hacer el match con el Snapshot', () => {
          const valor = 100;
          const wrapper = shallow(<CounterApp/>);
          expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de dar el valor de 100', ()=> {
        const value="100";
        const wrapper = shallow(<CounterApp value={100} />);
        const valorHtml = wrapper.find('h2').text();
        expect( valorHtml ).toBe(value);
        console.log( value );
    });
});
