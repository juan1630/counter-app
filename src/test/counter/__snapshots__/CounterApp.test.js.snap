// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Son las pruebas del componente <CounterApp/> Debe de hacer el match con el Snapshot 1`] = `
<Fragment>
  <h1>
     Counter App 
  </h1>
  <h2>
    2
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
  <button
    onClick={[Function]}
  >
     RESET 
  </button>
</Fragment>
`;
